### PR TITLE
Add notebook-server-jupyter-pytorch-full dockerfiles and CI/CD

### PR DESCRIPTION
**Pull request by DavidSpek**
This PR adds the PyTorch Jupyter opinionated images from #5582 that will serve as a starting point for (new) users and projects by adding an opinionated set of common tools. It uses https://github.com/kubeflow/kubeflow/pull/5626 as the base image. These images also serve as important examples for people that would like to customize the PyTorch base image for their distinct needs. More details about the downstream use of this image can be seen in the PR linked above. 

Initial images were built and pushed to test CI/CD:
public.ecr.aws/j1r0q0g6/notebooks/notebook-servers/jupyter-pytorch-full:master-28af7716
public.ecr.aws/j1r0q0g6/notebooks/notebook-servers/jupyter-pytorch-cuda-full:master-28af7716

/cc @kubeflow/wg-notebooks-leads @thesuperzapper

### CODE REVIEW

File: py/kubeflow/kubeflow/ci/workflow_utils.py [hunk 1]
```diff
@@ -200,14 +202,17 @@ def create_kaniko_task(self, task_template, dockerfile, context,
...
 
         # append the tag base-commit[0:7]
         if ":" not in destination:
             sha = os.getenv("PULL_BASE_SHA", "12341234kanikotest")
             base = os.getenv("PULL_BASE_REF", "master")
             destination += ":%s-%s" % (base, sha[0:8])
 
-        kaniko["name"] = "kaniko-build-push"
+        # add short UUID to step name to ensure it is unique
+        kaniko["name"] = "kaniko-build-push-" + ''.join(random.choices(alphabet, k=8))
```
Comment (kimwnasptd) [inline py/kubeflow/kubeflow/ci/workflow_utils.py:215]: Could you provide some more details as to why this change is needed?

Comment (DavidSpek) [reply py/kubeflow/kubeflow/ci/workflow_utils.py:215]: Otherwise there are 2 steps in the workflow (this workflow builds 2 images) that are both named `kaniko-build-push`. I wasn't sure if it would like that so I tested it and Argo does not :).

Comment (DavidSpek) [reply py/kubeflow/kubeflow/ci/workflow_utils.py:215]: The same thing is done here BTW: https://github.com/kubeflow/kubeflow/pull/5626/files#diff-a58831ea15b3ba92fe93b2196dc2867189dc87c3836c617833dbdf652ad22fd0R205

Comment (DavidSpek) [reply py/kubeflow/kubeflow/ci/workflow_utils.py:215]: Here you can see what the workflow I just linked looks like when it runs. 
https://argo.kubeflow-testing.com/workflows/kubeflow-test-infra/kubeflow-kubeflow-presubmit-nb-j-pt-5626-5feb726-6224-63fc?tab=workflow&nodeId=kubeflow-kubeflow-presubmit-nb-j-pt-5626-5feb726-6224-63fc-2134638194

Comment (kimwnasptd) [reply py/kubeflow/kubeflow/ci/workflow_utils.py:215]: Makes sense

**Comment by thesuperzapper** [comment]
Looks good!
/approve
/lgtm
